test/PrintAsObjC/compatibility-symbols.swift

```diff
@@ -0,0 +1,11 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend(mock-sdk: %clang-importer-sdk) %S/../Inputs/empty.swift -typecheck -emit-objc-header-path %t/empty.h
+// RUN: %clang -extract-api -o %t/empty.symbols.json --target=%target-triple -isysroot %clang-importer-sdk-path -F %clang-importer-sdk-path/frameworks --extract-api-ignores=%swift_obj_root/share/swift/compatibility-symbols -fmodules -x objective-c-header %t/empty.h
```

Comment:
If I got that right you run clang extract-api on a compatibility header and check that nothing ends up in the output? I still think the compatibility header has a module name dependent macro in it?

Reply:
That's correct. The header guard macro is being filtered out here by clang's own filtering. It's not a problem here, but if we were to (e.g.) run extract-api on a header that was merged together by Xcode, it would be a problem there. The merging behavior disrupts Clang's header-guard detection, because there the first thing in the file isn't the header-guard declaration, but a processor arch check. I'm not sure the best way to handle that, especially since all we are able to provide to Clang is a static file that we install in the toolchain.